frappe/public/js/frappe/form/sidebar/share.js

```diff
@@ -187,4 +191,52 @@ frappe.ui.form.Share = class Share {
 			});
 		});
 	}
+	async share_modal() {
+		const default_expiry = frappe.boot.sysdefaults.document_share_key_expiry;
+		const share_modal = new frappe.ui.Dialog({
+			title: __("Share Link"),
```

Comment:
I think we need to introduce the words **Public** and **Read Only** here to make it more explicit that this feature is to share a read-only print view to external users.